rocAL/source/decoders/audio/sndfile_decoder.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "sndfile_decoder.h"
+
+#include <commons.h>
+
+#include <cstdio>
+#include <cstring>
+
+#ifdef ROCAL_AUDIO
+
+SndFileDecoder::SndFileDecoder(){};
```

Comment:
I think, we need to have a better name than SndFileDecoder

Reply:
@rrawther  please let us know if we could change to SndFileAudioDecoder or GenericAudioDecoder

Reply:
Renamed SndFileDecoder to GenericAudioDecoder as suggested by Lakshmi in weekly call